### PR TITLE
Use errgroup to gracefully shutdown HTTP server in knockdown

### LINKED ISSUE #1751: Target Test Pod Exits Without Sending HTTP Response Status
**Describe the bug**
The target test pod will sometimes exit after successfully receiving the target event but will not send the response status causing the trigger to get an EOF error.

 **Expected behavior**
The target test pod should send status OK before exiting when receiving the target event.

 **To Reproduce**
This issue causes occasional failures in TestGCPBrokerMetrics.

### PR DESCRIPTION
**Pull request by ian-mi**
Fixes #1751

### CODE REVIEW

**Review by grac3gao — COMMENTED** [review]
/lgtm
/hold for comments from other reviewers

File: test/test_images/internal/knockdown/knockdown.go [hunk 1]
```diff
@@ -39,92 +44,84 @@ type Config struct {
...
+			if err != nil {
+				return err
+			}
+			e, err := binding.ToEvent(ctx, msg)
+			if err != nil {
+				msg.Finish(err)
+				return err
+			}
+			if kdr.Knockdown(*e) {
+				done = true
```
Comment (Harwayne) [inline test/test_images/internal/knockdown/knockdown.go:86]: Why set `done` here, rather than:
```golang
pcancel()
msg.Finish(nil)
return nil
```

Comment (ian-mi) [reply test/test_images/internal/knockdown/knockdown.go:86]: The receiver can't return here as the server may accept additional requests before shutting down. In order for the server to shutdown gracefully those requests need to be processed so the receiver will continue running until it gets an EOF.

Comment (Harwayne) [reply test/test_images/internal/knockdown/knockdown.go:86]: Please add a comment to that affect. Essentially stating why we need the `done` variable.

**Pull request merged**